### PR TITLE
Require experimental design with filenames to only have string type as input

### PR DESCRIPTION
**Pull request by jpfeuffer**
As requested on Slack. To be compatible with nf-tower.
This will lead to fewer errors but quite a bit more work for users.

Edit: More explanation:
One problem is that with the nf-core param schema, parameters need to have a specific type. String or List, where List isn't even supported yet. Therefore, the input parameter should not change the meaning in different "modes" of the workflow.
This is fixed now by always requiring an experimental design, from which also file locations are read. This can be an SDRF with parameters or a plain OpenMS experimental design. A root_folder and different filetype can be chosen in the "Staging" parameters to account for people using an experimental design after moving/downloading remote files or converting them. (this was possible before for SDRFs already).

I think this makes the interface a bit cleaner and allows full compatibility with all nf-core/nf-tower functionality.
However it is a bit more work for users. They can use the "create_simple_design.py" script for example and then edit in Excel or something.

### CODE REVIEW

**Comment by jpfeuffer** [comment]
Major blocker: https://github.com/OpenMS/OpenMS/issues/5556
Need to update conda and the container afterwards.

File: conf/test.config [hunk 1]
```diff
@@ -16,16 +16,8 @@ params {
   max_time = 48.h
 
   // Input data
-  input = [
```
Comment (timosachsenberg) [inline conf/test.config:19]: what changed here? now optional?

File: nextflow_schema.json [hunk 2]
```diff
@@ -17,8 +17,8 @@
...
+                    "description": "URI/path to an [SDRF](https://github.com/bigbio/proteomics-metadata-standard/tree/master/annotated-projects) file (with ending .sdrf or .sdrf.tsv) **OR** a tab-separated experimental design file (.tsv) in OpenMS' own [format](http://www.openms.de/doxygen/release/latest/html/classOpenMS_1_1ExperimentalDesign.html#details). All input files need to be specified with full paths in the corresponding columns. Those can be any URIs or local paths with schemata supported by nextflow (e.g. http/ftp/s3)",
```
Comment (timosachsenberg) [inline nextflow_schema.json:20]: ok so passing mzMLs directly doesn't work anymore right?

**Review by timosachsenberg — APPROVED** [review]
looks good. maybe add some comment to the PR about what changed regarding functionality (e.g., passing mzML files as input etc.)

**Comment by jpfeuffer** [comment]
Yes exactly, so one problem is that with the nf-core param schema, parameters need to have a specific type. String or List, where List isn't even supported yet. Therefore, the input parameter should not change the meaning in different "modes" of the workflow.
This is fixed now by always requiring an experimental design, from which also file locations are read. This can be an SDRF with parameters or a plain OpenMS experimental design. A root_folder and different filetype can be chosen in the "Staging" parameters to account for people using an experimental design after moving/downloading remote files or converting them. (this was possible before for SDRFs already).

I think this makes the interface a bit cleaner and allows full compatibility with all nf-core/nf-tower functionality.
However it is a bit more work for users. They can use the "create_simple_design.py" script for example and then edit in Excel or something.

File: nextflow_schema.json (hunk 2 above)
Comment (jpfeuffer) [reply nextflow_schema.json:20]: correct.

**Comment by timosachsenberg** [comment]
ok sounds good. will probably break some user workflows but well... probably needed

**Comment by jpfeuffer** [comment]
We can increase the major version for the next release to represent that. Not sure if this is enough to justify a major version.

File: conf/test.config (hunk 1 above)
Comment (jpfeuffer) [reply conf/test.config:19]: no they are included in the design now

**Pull request merged**